include/swift/Remote/MetadataReader.h

```diff
@@ -992,6 +992,35 @@ class MetadataReader {
     swift_runtime_unreachable("Unhandled IsaEncodingKind in switch.");
   }
 
+  /// Read the offset of the generic parameters of a class from the nominal
+  /// type descriptor. If the class has a resilient superclass, we also
+  /// have to read the superclass size and add that to the offset.
+  ///
+  /// The offset is in units of words, from the start of the class's
+  /// metadata.
+  std::pair<bool, uint32_t>
+  readGenericArgsOffset(MetadataRef metadata,
+                        NominalTypeDescriptorRef descriptor) {
+    if (metadata->getKind() == MetadataKind::Class) {
+      auto *classMetadata = cast<TargetClassMetadata<Runtime>>(metadata);
```

Comment:
If we support ``cast<>``, I'm sure we also support ``dyn_cast<>``.

Reply:
Thanks.